internal/sourcemap/metadata_fetcher.go

```diff
@@ -101,9 +101,9 @@ func (s *MetadataESFetcher) startBackgroundSync(parent context.Context) {
 	go func() {
 		s.logger.Debug("populating metadata cache")
 
-		ctx, cancel := context.WithTimeout(parent, 1*time.Second)
-		err := s.ping(ctx)
-		cancel()
+		// the parent ctx does not have a deadline so we
+		// rely on the es client timeout (default 5s).
+		err := s.ping(parent)
```

Comment:
Do we still need the ping? Can we get rid of the the first run code, and just keep the loop  below and close `s.init` on the first successful call to `s.sync`?

Like this:

```go
func (s *MetadataESFetcher) ready() <-chan struct{} {                        
        return s.readyChan                                                   
}                                                                            

func (s *MetadataESFetcher) startBackgroundSync(ctx context.Context) {
        go func() {
                defer close(s.invalidationChan)
                defer s.logger.Info("sourcemap metadata update routine done")
                s.logger.Debug("populating sourcemap metadata cache")
                                                                             
                ticker := time.NewTicker(30 * time.Second)
                defer ticker.Stop()
                first := make(chan struct{})
                close(first)

                var setAvailable bool
                for {
                        select {
                        case <-ctx.Done():
                                return
                        case <-first:
                                first = nil
                        case <-ticker.C:
                        }
                        if err := s.sync(ctx); err != nil {
                                s.logger.With(
                                        logp.Error(err),
                                ).Error("failed to sync sourcemaps metadata")
                                continue
                        }
                        if !setAvailable {
                                setAvailable = true
                                close(s.readyChan)
                        }
                }
        }()
}
```

Reply:
I got rid of the ping and the arbitrary 10s timeout. Sync was update to return errFetcherUnavailable on err.